sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/tests/ScenarioTests/SqlResourcesOperationsTests.cs

```diff
@@ -246,6 +255,120 @@ public void SqlCRUDTests()
             }
         }
 
+        [Fact]
+        public void SqlRoleTests()
+        {
+            var handler1 = new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK };
+            using (MockContext context = MockContext.Start(this.GetType()))
+            {
+                // Create client
+                CosmosDBManagementClient cosmosDBManagementClient = CosmosDBTestUtilities.GetCosmosDBClient(context, handler1);
+
+                SqlRoleDefinitionCreateUpdateParameters sqlRoleDefinitionCreateUpdateParameters = new SqlRoleDefinitionCreateUpdateParameters
+                {
+                    RoleName = "roleName",
+                    Type = RoleDefinitionType.CustomRole,
+                    AssignableScopes = new List<string>
+                    {
+                        string.Format("/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.DocumentDB/databaseAccounts/{2}", cosmosDBManagementClient.SubscriptionId, resourceGroupName, databaseAccountName)
+                    },
+                    Permissions = new List<Permission>
+                    {
+                        new Permission
+                        {
+                            DataActions = new List<string>
+                            {
+                                "Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/items/create",
+                                "Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/items/read",
+                                "Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/items/delete",
+                                "Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/items/replace"
+                            }
+                        }
+                    }
+                };
+
+                SqlRoleDefinitionGetResults sqlRoleDefinitionGetResults = cosmosDBManagementClient.SqlResources.CreateUpdateSqlRoleDefinitionWithHttpMessagesAsync(roleDefinitionId, resourceGroupName, databaseAccountName2, sqlRoleDefinitionCreateUpdateParameters).GetAwaiter().GetResult().Body;
+                Assert.NotNull(sqlRoleDefinitionGetResults);
+                Assert.Equal(roleDefinitionId, sqlRoleDefinitionGetResults.Name);
+
+                SqlRoleDefinitionGetResults sqlRoleDefinitionGetResults2 = cosmosDBManagementClient.SqlResources.GetSqlRoleDefinitionWithHttpMessagesAsync(roleDefinitionId, resourceGroupName, databaseAccountName2).GetAwaiter().GetResult().Body;
+                Assert.NotNull(sqlRoleDefinitionGetResults2);
+                Assert.Equal(roleDefinitionId, sqlRoleDefinitionGetResults2.Name);
+
+                VerifyEqualSqlRoleDefinitions(sqlRoleDefinitionGetResults, sqlRoleDefinitionGetResults2);
```

Comment:
We should verify equality against SqlRoleDefinitionCreateUpdateParameters instead of an earlier GET result. The current equality check isn't meaningful.

Reply:
Ok I've added VerifyCreateUpdateRoleDefinition/Assignment to verify against the CreateUpdateParameters object. I'm leaving this VerifyEqualSqlRoleDefinitions call because it compares the Create with the Get